src/bundler.ts

```diff
@@ -150,31 +179,64 @@ class Bundler {
         await analyzer._analyzeResolved(resolved);
     const documentAst = dom5.parseFragment(backingDocument.document.contents);
     this.pathResolver.resolvePaths(documentAst, resolved, documentUrl);
-    const importParent = htmlImport.parentNode;
+    let importParent;
+    if (matchers.afterHiddenDiv(htmlImport)) {
```

Comment:
This function feels too long. I'd like to see the hidden div logic split out to make it clearer what is pure inlining vs our hacks.

Also, splitting out the hidden div functionality should enable some highly needed unit tests around it.